app/parsers/ip.py

```diff
@@ -31,31 +31,20 @@ def _is_valid_ip(raw_ip):
             return False
         return True
 
-    @staticmethod
-    def _whitelist_ip(raw_ip, whitelist):
-        try:
-            ip = ip_address(raw_ip)
-        except BaseException:
-            return None
-        if 'multicast' not in whitelist:
-            if ip.is_multicast:
-                return None
-        if 'loopback' not in whitelist:
-            if ip.is_loopback:
-                return None
-        if 'link_local' not in whitelist:
-            if ip.is_link_local:
-                return None
-        if 'reserved' not in whitelist:
-            if ip.is_reserved:
-                return None
-        if 'global' not in whitelist:
-            if ip.is_global:
-                return None
-        if 'unspecified' not in whitelist:
-            if ip.is_unspecified:
-                return None
-        if 'private' not in whitelist:
-            if ip.is_private:
-                return None
-        return str(ip)
+    whitelist_options = {

```

Comment:
move to line top ~line 7

Reply:
that is where i had it at first for convention, but thought it was easier to read here, easy fix